python-sdk/src/astro/databases/base.py

```diff
@@ -195,14 +195,16 @@ def default_metadata(self) -> Metadata:
         """
         raise NotImplementedError
 
-    def populate_table_metadata(self, table: BaseTable) -> BaseTable:
+    def populate_table_metadata(self, table: BaseTable, source_table: None | BaseTable = None) -> BaseTable:
```

Comment:
This makes it pluggable, so that we can change different table attribtues for different databases  with different logic